### PR TITLE
[docs] Clean up docs to account for expo-permissions deprecation

### PR DESCRIPTION
**Pull request by brentvatne**
# Why

SDK 41 deprecates expo-permissions, so we should not mention it anywhere outside of its own API reference page, alongside a deprecation warning. 

Fixes ENG-661

# How

- Remove *most* references to expo-permissions outside of the Permissions API reference page. I wasn't able to remove links to `PermissionsResponse` type because then we'd have to inline it on every page -- I'm open to ideas for how to handle this.
  - edit: as we switch over to auto-generated docs, we can just re-export PermissionsResponse from every module that uses it and it'll be included in the API reference on that page, such as with Pedometer on SDK 41 now
- I copied guides type of content out of the API reference and over to the `guides` section, but left the same content on the API reference page for now because I want to double check with @byCedric that it is safe to remove (not sure where we're linking to it? maybe we need to set up some nice redirects).
- I left the sensors stuff alone for now, we should address this after https://github.com/expo/expo/pull/12501 lands.

# Test Plan

"RTFM"

### CODE REVIEW

File: docs/pages/versions/v41.0.0/sdk/permissions.md [hunk 1]
```diff
@@ -6,6 +6,8 @@ sourceCodeUrl: 'https://github.com/expo/expo/tree/sdk-41/packages/expo-permissio
 import InstallSection from '~/components/plugins/InstallSection';
 import PlatformsSection from '~/components/plugins/PlatformsSection';
 
+> **expo-permissions is deprecated.** Use permissions getters and requesters on specific modules instead, such as [MediaLibrary.getPermissionsAsync()](../media-library.md/#medialibrarygetpermissionsasync) and [MediaLibrary.requestPermissionsAsync()](../media-library.md/#medialibraryrequestpermissionsasync)..
```
Comment (cruzach) [inline docs/pages/versions/v41.0.0/sdk/permissions.md:9]:
```suggestion
> **expo-permissions is deprecated.** Use permissions getters and requesters in specific modules instead, such as [MediaLibrary.getPermissionsAsync()](../media-library.md/#medialibrarygetpermissionsasync) and [MediaLibrary.requestPermissionsAsync()](../media-library.md/#medialibraryrequestpermissionsasync)..
```

File: docs/pages/versions/v41.0.0/sdk/permissions.md [hunk 2]
```diff
@@ -185,6 +189,8 @@ async function checkMultiPermissions() {
 
 ### `Permissions.askAsync(...types)`
 
+> **Deprecated.** Use permissions requesters on specific modules instead, such as [MediaLibrary.requestPermissionsAsync()](../media-library.md/#medialibraryrequestpermissionsasync).
```
Comment (cruzach) [inline docs/pages/versions/v41.0.0/sdk/permissions.md:192]:
```suggestion
> **Deprecated.** Use permissions requesters in specific modules instead, such as [MediaLibrary.requestPermissionsAsync()](../media-library.md/#medialibraryrequestpermissionsasync).
```

File: docs/pages/guides/permissions.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+In Expo Go, there isn't much you need to think about to interact with permissions beyond requesting permissions before using certain APIs. This changes when you want to deploy your app to an app store. Please read the [permissions on iOS](#permissions-on-ios) and [permissions on Android](#permissions-on-android) sections carefully before deploying your app to stores. If you don't configure or explain the permissions properly **it may result in your app getting rejected or pulled from the stores**. Read more about deploying to the stores in the [App Store Deployment Guide](../../../distribution/app-stores.md#system-permissions-dialogs-on-ios).
+
+## iOS
+
+### Managed workflow
+
+To request permissions on iOS, you have to describe why the permissions are requested and [install the library](#permissions-and-required-packages-on-ios) that can request this permission. In the managed workflow, you can do that by customizing the `ios.infoPlist` property in your [`app.json` file](../../../workflow/configuration.md#ios). When using the bare workflow, you have to edit the `info.plist` file directly.
```
[the remark this answers is not in the document]
Comment (cruzach) [reply docs/pages/guides/permissions.md:13]: Also it's a little odd to reference bare workflow in the Managed workflow section when bare workflow has its own section up next

**Review by cruzach — COMMENTED** [review]
> I wasn't able to remove links to PermissionsResponse type because then we'd have to inline it on every page -- I'm open to ideas for how to handle this.

I think I prefer inlining on each page tbh- always nicer to not have to navigate to a new page to see what a function is going to return. And probably not much work now that we're moving to autogeneration of the API docs

File: docs/pages/guides/permissions.md (hunk 3 above)
Comment (brentvatne) [inline docs/pages/guides/permissions.md:13]:
```suggestion
To request permissions on iOS, you have to describe why the permissions are requested and [install the library](#permissions-and-required-packages-on-ios) that can request this permission. In the managed workflow, you can do that by customizing the `ios.infoPlist` property in your [`app.json` file](../../../workflow/configuration.md#ios).
```

File: docs/pages/guides/permissions.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+## iOS
+
+### Managed workflow
+
+To request permissions on iOS, you have to describe why the permissions are requested and [install the library](#permissions-and-required-packages-on-ios) that can request this permission. In the managed workflow, you can do that by customizing the `ios.infoPlist` property in your [`app.json` file](../../../workflow/configuration.md#ios).
+
+See the [`Permission types`](#permission-types) below to learn about what `infoPlist` property you need for each permission. You can find the full list of available properties in [Apple's InfoPlistKeyReference](https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CocoaKeys.html#//apple_ref/doc/uid/TP40009251-SW1). Apple also documents the basic guidelines for the structure of the message in the [Human Interface Guidelines](https://developer.apple.com/design/human-interface-guidelines/ios/app-architecture/requesting-permission/).
```
Comment (brentvatne) [inline docs/pages/guides/permissions.md:15]:
```suggestion
You can find the full list of available properties in [Apple's InfoPlistKeyReference](https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CocoaKeys.html#//apple_ref/doc/uid/TP40009251-SW1). Apple also documents the basic guidelines for the structure of the message in the [Human Interface Guidelines](https://developer.apple.com/design/human-interface-guidelines/ios/app-architecture/requesting-permission/).
```

File: docs/pages/guides/permissions.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+> **Note:** apps using permissions without descriptions _may be rejected from the App Store_. (see the [App Store Deployment Guide](../../../distribution/app-stores.md#system-permissions-dialogs-on-ios))
+
+### Bare workflow
+
+To request permissions on iOS, you have to describe why the permissions are requested and [install the library](#permissions-and-required-packages-on-ios) that can request this permission. When using the bare workflow, you have to edit the project `Info.plist`.
```
Comment (brentvatne) [inline docs/pages/guides/permissions.md:21]:
```suggestion
To request permissions on iOS, you have to describe why the permissions are requested and install the library that requests and uses the permission. When using the bare workflow, you have to edit the project `Info.plist`.
```

File: docs/pages/guides/permissions.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+See the [`Permission types`](#permission-types) below to learn about what `Info.plist` property you need for each permission. You can find the full list of available properties in [Apple's InfoPlistKeyReference](https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CocoaKeys.html#//apple_ref/doc/uid/TP40009251-SW1). Apple also documents the basic guidelines for the structure of the message in the [Human Interface Guidelines](https://developer.apple.com/design/human-interface-guidelines/ios/app-architecture/requesting-permission/).
```
Comment (brentvatne) [inline docs/pages/guides/permissions.md:23]:
```suggestion
You can find the full list of available properties in [Apple's InfoPlistKeyReference](https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CocoaKeys.html#//apple_ref/doc/uid/TP40009251-SW1). Apple also documents the basic guidelines for the structure of the message in the [Human Interface Guidelines](https://developer.apple.com/design/human-interface-guidelines/ios/app-architecture/requesting-permission/).
```

**Comment by brentvatne** [comment]
> > I wasn't able to remove links to PermissionsResponse type because then we'd have to inline it on every page -- I'm open to ideas for how to handle this.
> 
> I think I prefer inlining on each page tbh- always nicer to not have to navigate to a new page to see what a function is going to return. And probably not much work now that we're moving to autogeneration of the API docs

great, this will be handled automatically as part of docs generation process, we just need to be sure to export `PermissionsRequest` from any module that uses it, cc @Simek

**Pull request merged**